File: app/src/main/java/com/unsplash/client/fragment/injectors/PhotoInjector.java
package com.unsplash.client.fragment.injectors;


import android.widget.ImageView;

import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;
import com.unsplash.client.GlideApp;
import com.unsplash.client.R;
import com.unsplash.client.data.model.PhotoModel;

import net.idik.lib.slimadapter.SlimInjector;
import net.idik.lib.slimadapter.viewinjector.IViewInjector;

public class PhotoInjector implements SlimInjector<PhotoModel> {

    @Override
    public void onInject(PhotoModel data, IViewInjector injector) {


        injector
                .with(R.id.image, view -> {

                    GlideApp.with(view.getContext())
                            .load(data.getUrlsPhoto().getRegular())
                            .apply(new RequestOptions().centerCrop())
                            .diskCacheStrategy(DiskCacheStrategy.ALL)
                            .into((ImageView)view); })
                .text(R.id.description, data.getDescription());


    }
}

File: app/src/main/java/com/unsplash/client/fragment/injectors/PhotoInjector.kt
package com.unsplash.client.fragment.injectors

import android.view.View
import android.widget.ImageView

import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.RequestOptions
import com.unsplash.client.GlideApp
import com.unsplash.client.R
import com.unsplash.client.data.model.PhotoModel

import net.idik.lib.slimadapter.SlimInjector
import net.idik.lib.slimadapter.viewinjector.IViewInjector

class PhotoInjector : SlimInjector<PhotoModel> {

    override fun onInject(data: PhotoModel, injector: IViewInjector<*>) {

        injector
                .with<View>(R.id.image) {

                    GlideApp.with(it.context)
                            .load(data.urlsPhoto?.regular)
                            .apply(RequestOptions().centerCrop())
                            .diskCacheStrategy(DiskCacheStrategy.ALL)
                            .into(it as ImageView)
                }
                .text(R.id.description, data.description)


    }
}
